Issue 44632: Publish study wizard support for selecting user-defined Queries

cnathe opened the pull request:
#### Rationale
https://www.labkey.org/home/Developer/issues/issues-details.view?issueId=44632
See related PR for rationale. This PR fixes some existing automated tests that use the publish study wizard. I also refactored a few things to try and make the publish study wizard advancement easier (less duplicate locators in test code) via study helper.

#### Related Pull Requests
* https://github.com/LabKey/platform/pull/3034
* https://github.com/LabKey/testAutomation/pull/1004

cnathe committed:
Issue 44632: Publish study wizard test helper update for new Queries panel

cnathe committed:
refactor some study test helpers for advancing through the publish wizard
- for the common cases of changing nothing or selecting all in the grid

cnathe committed:
Merge branch 'develop' into fb_issue44632PublishQueries

cnathe committed:
StudyHelper publishStudy() fix to include prop to check all for advanceThroughPublishStudyWizard() calls

cnathe committed:
Timepoints panel still uses studyWizardVisitList cls

cnathe committed:
StudyPublishTest fix for study objects and folder objects select all

cnathe committed:
Fix for StudyPublishTest participant panel label check

cnathe committed:
StudyPublishQueriesTest - new test for queries selection in publish study wizard

labkey-tchad commented on src/org/labkey/test/util/StudyHelper.java:
```diff
@@ -389,6 +381,30 @@ public void publishStudy(String studyName, int expectedPipelineJobs, String subj
         _test.waitForPipelineJobsToComplete(expectedPipelineJobs, "Publish Study", false);
     }
 
+    public void advanceThroughPublishStudyWizard(List<StudyHelper.Panel> wizardPanels)
+    {
+        advanceThroughPublishStudyWizard(wizardPanels, false);
+    }
```
Add an overload that takes a single panel. Seems like a common use case.

```suggestion
    public void advanceThroughPublishStudyWizard(StudyHelper.Panel wizardPanel, boolean selectAll)
    {
        advanceThroughPublishStudyWizard(List.of(wizardPanels), selectAll);
    }

    public void advanceThroughPublishStudyWizard(List<StudyHelper.Panel> wizardPanels)
    {
        advanceThroughPublishStudyWizard(wizardPanels, false);
    }
```

cnathe replied:
Good call. Done.

labkey-tchad commented on src/org/labkey/test/util/StudyHelper.java:
```diff
@@ -566,4 +582,40 @@ public boolean isSpecimenModuleActive()
         VIAL,
         SPECIMEN
     }
+
+    public enum Panel
+    {
+        studyGeneralSetup("General Setup", null),
+        studyWizardParticipantList("Participants", "Participant Group"),
+        studyWizardMouseList("Mice", "Mouse Group"),
+        studyWizardDatasetList("Datasets", "Label"),
+        studyWizardVisitList("Visits", "Label"),
```
If you make this enum extend an interface, it could support custom participant types (e.g. Pandas).
```suggestion
    public interface Panel
    {
        String getPanelTitle();
        String getLabelColumn();
    }

    public enum Panels implements Panel
    {
        studyGeneralSetup("General Setup", null),
        studyWizardParticipantList("Participants", "Participant Group"),
        studyWizardMouseList("Mice", "Mouse Group"),
        studyWizardDatasetList("Datasets", "Label"),
        studyWizardVisitList("Visits", "Label"),
```

Then maybe add a convenience method for participants:
```
    public static Panel participantList(String singular, String plural)
    {
        return new Panel()
        {
            @Override
            public String getPanelTitle()
            {
                return plural;
            }

            @Override
            public String getLabelColumn()
            {
                return singular + " Group";
            }
        };
    }
```

cnathe replied:
Done.

cnathe committed:
Merge branch 'develop' into fb_issue44632PublishQueries

cnathe committed:
Code review feedback
- StudyHelper.participantList()
- advanceThroughPublishStudyWizard() overload which takes single panel

cnathe committed:
missed from previous commit
